service/middleware.go

```diff
@@ -86,3 +81,45 @@ func CORSMiddleware() func(http.Handler) http.Handler {
 		return handlers.CORS(headers, origins, methods, exposed, credentials)(h)
 	}
 }
+
+// JWTValidate verifies that the JWT token is valid. The token is set after logging-in via Openid
+func JWTValidate(protected http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+		// Do not validate access if user authentication is off (default)
+		if !datastore.Environ.Config.EnableUserAuth {
+			protected.ServeHTTP(w, r)
+			return
+		}
+
+		// Get the JWT from the header or cookie
+		jwtToken, err := usso.JWTExtractor(r)
+		if err != nil {
+			log.Println(err.Error())
+			// TODO: don't leave pages unprotected when there is no token
```

Comment:
Yes - I imagine that it would redirect to / with an error message, but I'm open to suggestions... when we come to that.